### DISCUSSION TITLE
DIDs Provide an Identity Metasystem

### DISCUSSION THREAD

**Discussion by TelegramSam** [trustoverip/trust-spanning-protocol#36] [General]
The DID Core Spec isn't just an identifier standard, it's an Identifier Metasystem. Each method is a participant within that metasystem.

If we don't use the DID identifier metasystem, we'll need to create our own, solving the same issues the spec does. We may be able to make some slight improvements, but will those improvements be worth the effort to reinvent?

It's worth noting that support of an identifier metasystem doesn't require supporting all participants in that metasystem. We don't need to support all DID Methods. Instead, we can select only the ones with the properties we insist upon. This can be done without inventing our own identity metasystem.

We should adopt DIDs as our Identity Metasystem and move along.

Thoughts?

**Comment by dhh1128** [comment]
Hey, @TelegramSam , I saw your other discussion, but missed this one until just now. Sorry.

I believe one of the features you are thinking we'd have to reinvent include how we advertise and discover our roots of trust (what many DID methods get by pointing to a particular blockchain instance and what KERI would get by building and advertising a watcher network or uber watcher). Once we had such a mechanism, we'd also have to describe how it is governed; that's another point of invention. Am I right? Another might be how we build URLs that are namespaced by the identifier and followed by other stuff of interest?

What other items did you have in mind?

**Reply by TelegramSam** [reply]
Top of mind identity metasystem features that the DID Core spec has:
- Standard way to combine identifiers by namespace
- Requirements for DID Methods that detail how to resolve identifiers
- Standard expectations about what you get when you resolve an identifier

You hint at governance and other features, though those are mostly left to the DID Methods.

I need to clarify my point a little: We do not need to support every DID method to adopt the DID identifier metasystem. We can and should select which DID Methods contain the properties we need. My main point is that we should require use within the DID metasystem, instead of defining Yet Another Metasystem that contains DIDs and other things.

I think we should settle on DIDs, and select the subset of DID Methods that provide the properties we need.

**Reply by talltree** [reply]
Trying to put aside my bias from having spent 7 years of my life (well, not continuously) on the DID 1.0 spec, I find @TelegramSam's argument compelling. The truth is that there is a fairly easy path for `verifiable identifiers` (VIDs) such as HTTPS URLs to be expressed as a `DID method` (that's exactly what the `did:web` method does), and the benefit of doing that is that it standardizes exactly the discovery/resolution properties that Sam lists above (and that we need to make it work with the `ToIP stack`).

So I could see revising the [ToIP Technology Architecture Specification V1.0](https://github.com/trustoverip/TechArch/blob/main/spec.md) (TAS) requirements for `ToIP identifiers` to keep the term VIDs, but specify that for interoperability reasons, a VID MUST be a DID method that satisfies certain requirements (TBD), and that SHOULD be an AID (which we say already).

Thoughts from others on this potential revision to the TAS requirements?
